From a21b10086fa2d788aee789c704b6e2a08435d4d6 Mon Sep 17 00:00:00 2001
From: Charitardha <satya.charitardha.jayanti@intel.com>
Date: Thu, 29 Nov 2012 14:49:56 +0530
Subject: [PORT FROM MAIN] The sound of SMS only output from HDTV not from device when playing 3G AV streaming

BZ: 56007

In AudioPolicyManagerBase::getDeviceForStrategy(), the device should
be selected based on STRATEGY_SONIFICATION irrespective of whether
music stream is active or not when WiDi is connected. Then, sms
alert will be heard on both HDTV through WiDi and device.

Change-Id: I871ebdecf3a5a3cb0a39309c933a8f512d4590ed
Orig-Change-Id: I5fc44755555258d51d5119ca2e1283ca5eeae83e
Signed-off-by: Charitardha <satya.charitardha.jayanti@intel.com>
Reviewed-on: http://android.intel.com:8080/82051
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 255fe18..52d89ab 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -2124,8 +2124,12 @@ audio_devices_t AudioPolicyManagerBase::getDeviceForStrategy(routing_strategy st
             //   if it's for local playback. Note also that we use the same delay between both tests
             device = getDeviceForStrategy(STRATEGY_SONIFICATION, false /*fromCache*/);
         } else if (isStreamActive(AudioSystem::MUSIC, SONIFICATION_RESPECTFUL_AFTER_MUSIC_DELAY)) {
-            // while media is playing (or has recently played), use the same device
+            // while media is playing (or has recently played),
+            // use the same device if device is not WIDI,
+            // otherwise fall back on the sonification behavior
             device = getDeviceForStrategy(STRATEGY_MEDIA, false /*fromCache*/);
+            if (device == AudioSystem::DEVICE_OUT_WIDI)
+                device = getDeviceForStrategy(STRATEGY_SONIFICATION, false /*fromCache*/);
         } else {
             // when media is not playing anymore, fall back on the sonification behavior
             device = getDeviceForStrategy(STRATEGY_SONIFICATION, false /*fromCache*/);
-- 
1.7.4.1

